### DISCUSSION TITLE
"AdGuard Tracking Protection" breaks caching

### DISCUSSION THREAD

**Discussion by redjard** [uBlockOrigin/uAssets#21842] [General] [answered]
### Prerequisites

- [X] This is **NOT** a [YouTube](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [Facebook](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [Twitch](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [X] I did not answer truthfully to **ALL** the above checkpoints.

### URL(s) where the issue occurs.

\*  
literally all urls, except those that prevent caching already.

### Description

When activating "AdGuard Tracking Protection", no cache entries are generated. This leads to pages reloading when going back and forth in tab history, and when clicking recently visited links. This can be "fixed" by setting `cacheControlForFirefox1376932` to `unset` or `no-cache` in advanced settings.

This list is not (no longer?) by default enabled, but the issue is severe, so I hope this is ok to raise here.

### Other extensions used

none

### Configuration

<details>

```yaml
uBlock Origin: 1.55.0
Firefox: 121
filterset (summary):
 network: 83503
 cosmetic: 289
 scriptlet: 969
 html: 0
listset (total-discarded, last-updated):
 removed:
  user-filters: null
  ublock-filters: null
  ublock-badware: null
  ublock-privacy: null
  ublock-unbreak: null
  ublock-quick-fixes: null
  easylist: null
  easyprivacy: null
  urlhaus-1: null
  plowe-0: null
 added:
  adguard-spyware: 84923-150, 2m
filterset (user): [empty]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 184 ms
 maxAssetCacheWait: 11 ms
 cacheBackend: indexedDB
```
</details>

**Comment by krystian3w** [comment]
List is optional, maybe again will be deleted to 2-3 weeks until AdGuard Team remove problematic filters.

**Comment by gorhill** [comment]
It's that filter:

    ://*.$document,permissions=join-ad-interest-group=()\, run-ad-auction=()\, browsing-topics=()

It shouldn't apply to Firefox since it does not support these permissions. "uBlock filters -- Privacy" already block these, but only for Chromium-based browsers.

---

Since it's not a default list, issues arising from these are not addressed in uBO lists.

---

The following filter can be used to disable the AdGuard filter above:

    ://*.$document,permissions=join-ad-interest-group=()\, run-ad-auction=()\, browsing-topics=(),badfilter

**Comment by gorhill** [comment]
> no cache entries are generated

To be clear, only the HTML document is not cached, secondary resources are cached.

**Comment by ghajini** [comment]
@Alex-302

**Comment by Yuki2718** [comment, marked as answer]
https://github.com/AdguardTeam/AdguardFilters/commit/1873f8da3afa91ceec27b8f7991c15f13b5b5838
